src/lib/mongoBackend/mongoUpdateSubscription.cpp

```diff
@@ -355,14 +355,46 @@ static void setCondsAndInitialNotify
       status = subOrig.hasField(CSUB_STATUS)? getStringFieldF(subOrig, CSUB_STATUS) : STATUS_ACTIVE;
     }
 
-    std::string url;
+    HttpInfo httpInfo;
     if (subUp.notificationProvided)
     {
-      url = subUp.notification.httpInfo.url;
+      httpInfo = subUp.notification.httpInfo;
```

Comment:
Note that if notificationProvides is false, the information is taken from a BSONObj. Thus, using subUp.notification.httpInfo (a read-only field) does't work in all cases.